File: flink-code-java-0421/src/main/java/com/atguigu/day02/MapExample.java
package com.atguigu.day02;

import com.atguigu.day02.util.SensorReading;
import com.atguigu.day02.util.SensorSource;
import org.apache.flink.api.common.functions.MapFunction;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

public class MapExample {
    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.setParallelism(1);

        DataStreamSource<SensorReading> stream = env.addSource(new SensorSource());

        stream
                .map(r -> r.id)
                .print();

        stream
                .map(new MapFunction<SensorReading, String>() {
                    @Override
                    public String map(SensorReading sensorReading) throws Exception {
                        return sensorReading.id;
                    }
                })
                .print();

        stream
                .map(new IdExtractor())
                .print();

        env.execute();
    }

    public static class IdExtractor implements MapFunction<SensorReading, String> {
        @Override
        public String map(SensorReading sensorReading) throws Exception {
            return sensorReading.id;
        }
    }
}

File: flink-code-scala-0421/src/main/scala/com/atguigu/day02/MapExample.scala
package com.atguigu.day02

import com.atguigu.day02.util.{SensorReading, SensorSource}
import org.apache.flink.api.common.functions.MapFunction
import org.apache.flink.streaming.api.scala._

object MapExample {
  def main(args: Array[String]): Unit = {
    val env = StreamExecutionEnvironment.getExecutionEnvironment
    env.setParallelism(1)

    val stream = env.addSource(new SensorSource)

    stream
      .map(r => r.id)
      .print()

    stream
      .map(new MapFunction[SensorReading, String] {
        override def map(t: SensorReading): String = t.id
      })
      .print()

    stream
      .map(new MyMap)
      .print()

    env.execute()
  }

  class MyMap extends MapFunction[SensorReading, String] {
    override def map(t: SensorReading): String = t.id
  }
}